book/crud.rst

```diff
@@ -0,0 +1,767 @@
+Creating and Updating Entities
+------------------------------
+
+To be able to create new vehicles and update existing ones, you first have to create a form type
+and register it as a service:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Form/Type/VehicleType.php
+    namespace InventoryBundle\Form\Type;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+    class VehicleType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('model')
+                ->add('seats')
+                ->add('boughtAt')
+                ->add('leasedUntil')
+            ;
+        }
+
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'InventoryBundle\Entity\Vehicle',
+            ));
+        }
+
+        public function getName()
+        {
+            return 'inventory_vehicle';
+        }
+    }
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/InventoryBundle/Resources/config/form.yml
+    services:
+        inventory.form.type.vehicle:
+            class: InventoryBundle\Form\Type\VehicleType
+            tags:
+                - { name: form.type, alias: inventory_vehicle }
+
+Then, you will need to create the needed controller actions. You can simplify the actions if you
+create a dedicated method that is handling the form submission:
+
+.. code-block:: php
+    :linenos:
+
+    // src/InventoryBundle/Controller/VehicleController.php
+    namespace InventoryBundle\Controller;
+
+    // ...
+    use InventoryBundle\Entity\Vehicle;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class VehicleController extends Controller
+    {
+        /**
+         * @Route("/create", name="inventory.vehicle_create")
+         * @Template("InventoryBundle:Vehicle:update.html.twig")
+         * @Acl(
+         *     id="inventory.vehicle_create",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="CREATE"
+         * )
+         */
+        public function createAction(Request $request)
+        {
+            return $this->update(new Vehicle(), $request);
+        }
+
+        /**
+         * @Route("/update/{id}", name="inventory.vehicle_update", requirements={"id":"\d+"}, defaults={"id":0})
+         * @Template()
+         * @Acl(
+         *     id="inventory.vehicle_update",
+         *     type="entity",
+         *     class="InventoryBundle:Vehicle",
+         *     permission="EDIT"
+         * )
+         */
+        public function updateAction(Vehicle $vehicle, Request $request)
+        {
+            return $this->update($vehicle, $request);
+        }
+
+        private function update(Vehicle $vehicle, Request $request)
```

Comment:
Yes, you are totally right, it's deprecated since version 2.4, to be removed in 3.0. And though we are still at 2.3, I think we better stick to passing it as an argument as you have :+1: .